src/dagmc/tests/dagmc_pointinvol_test.cpp

```diff
@@ -0,0 +1,234 @@
+#include <gtest/gtest.h>
+
+#include "moab/Interface.hpp"
+#include "moab/Core.hpp"
+#include "DagMC.hpp"
+
+#include <iostream>
+
+using namespace moab;
+
+using moab::DagMC;
+
+moab::DagMC* DAG;
+
+static const char input_file[] = "test_geom.h5m";
+
+class DagmcPointInVolTest : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    // Create new DAGMC instance
+    DAG = new DagMC();
+    // Load mesh from file
+    rloadval = DAG->load_file(input_file);
+    assert(rloadval == moab::MB_SUCCESS);
+    // Create the OBB
+    rval = DAG->init_OBBTree();
+    assert(rval == moab::MB_SUCCESS);
+  }
+  virtual void TearDown() {
+    delete DAG;
+  }
+ protected:
+  moab::ErrorCode rloadval;
+  moab::ErrorCode rval;
+};
+
+TEST_F(DagmcPointInVolTest, dagmc_setup_test) {
+  ErrorCode rval = DAG->load_file(input_file);
+  EXPECT_EQ(rval, MB_SUCCESS);
+  rval = DAG->init_OBBTree();
+  EXPECT_EQ(rval, MB_SUCCESS);
+}
+
+TEST_F(DagmcPointInVolTest, dagmc_point_in) {
+  int result = 0;
+  int expected_result = 1;
+  double xyz[3] = {0.0, 0.0, 0.0};
+  int vol_idx = 1;
+  EntityHandle vol_h = DAG->entity_by_index(3, vol_idx);
+  ErrorCode rval = DAG->point_in_volume(vol_h, xyz, result);
+  EXPECT_EQ(rval, MB_SUCCESS);
+  EXPECT_EQ(expected_result, result);
+}
+
+int dagmc_point_in_vol_dir(double origin[3], double dir[3], int vol_idx) {
+  int result = 0;
+  EntityHandle vol_h = DAG->entity_by_index(3, vol_idx);
+  double xyz[3];
+  double next_surf_dist;
+  EntityHandle next_surf;
+
+  // normalise the vector
+  double dir_norm = (dir[0] * dir[0]) + (dir[1] * dir[1]) + (dir[2] * dir[2]);
+
+  dir[0] = dir[0] / sqrt(dir_norm);
```

Comment:
Best practices: I would create another `double[3]` here (maybe "dir_normalized"), rather than reassigning the parameter.  Additionally, based on the current usages of `dagmc_point_in_vol_dir`, it doesn't look like the callers expect the parameters to be modified.